File: tests/benchmarks/array_bench.py
import time

def array_test(n=5_000_000):
    start = time.perf_counter()
    arr = [i * 2 for i in range(n)]
    total = sum(arr)
    duration = time.perf_counter() - start
    print(f"Result: {total}")
    return duration

if __name__ == "__main__":
    print(f"Array Test: {array_test():.4f} sec")

File: tests/benchmarks/array_bench.rs
use std::time::Instant;

fn main() {
    let n = 5_000_000;
    let start = Instant::now();
    let mut arr = Vec::with_capacity(n);
    for i in 0..n {
        arr.push((i as i64) * 2);
    }
    let total: i64 = arr.iter().sum();
    let duration = start.elapsed();
    println!("Result: {}", total);
    println!("Array Test: {:.4} sec", duration.as_secs_f64());
}
